tmt/utils/git.py

```diff
@@ -26,6 +27,22 @@
     import tmt.base
 
 
+@dataclasses.dataclass
+class GitInfo:
+    """ Data container for commonly queried git data. """
+
+    git_root: Path
+    """Path to the git root."""
+    ref: str
+    """Most human-readable git ref."""
+    remote: str
+    """Git remote linked to the current git ref."""
+    default_branch: Optional[str]
+    """Default branch of the remote."""
+    url: Optional[str]
+    """Public url of the remote."""
```

Comment:
@happz Do you have near-term plans to use this one so that we cherry-pick it out, or shall we keep it in here?

Reply:
I don't follow, you mean this whole patch? It's supposed to land in the upcoming release, hopefully, there will be a swarm of reviewers arriving Soon(TM) :)

Reply:
I'm thinking specifically 29619fbf472e39ac58878a87489ca0aa9f1e7b2e if you would have plans to extend it, e.g. much of the `git_root`, `git_add` functions in there could be moved inside this, and then we would have a more minimal `import` for the git metadata. I also want to further `from_fmf_root` to `from_git_root` for probably more efficient caching, but I am not sure if I should add more here or postpone to a later PR to make this more easy to review.

Currently though, I am happy in the current form, probably it only lacks some release notes or other documentation?

Reply:
+1 for separate PR

Reply:
Yeah, +1 to a separate PR. Absolutely true there are several functions `git_something` that would deserve to have some kind of context to be attached to and to maintain.

Reply:
+1